Protocol/Tests/Protocol/Params/Param/Interprete/Exceptions/Exception/Display/CheckStateAttribute.cs

```diff
@@ -0,0 +1,162 @@
+{
+    using System;
+    using System.Collections.Generic;
+    using Skyline.DataMiner.CICD.Models.Protocol.Edit;
+    using Skyline.DataMiner.CICD.Models.Protocol.Enums;
+    using Skyline.DataMiner.CICD.Models.Protocol.Read;
+    using Skyline.DataMiner.CICD.Validators.Common.Interfaces;
+    using Skyline.DataMiner.CICD.Validators.Common.Model;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Common;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Common.Attributes;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Common.Extensions;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Generic;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Interfaces;
+
+    [Test(CheckId.CheckStateAttribute, Category.Param)]
+    internal class CheckStateAttribute : IValidate, ICodeFix, ICompare
+    {
+        // Please comment out the interfaces that aren't used together with the respective methods.
+
+        public List<IValidationResult> Validate(ValidatorContext context)
+        {
+            List<IValidationResult> results = new List<IValidationResult>();
+
+            foreach (IParamsParam param in context.EachParamWithValidId())
+            {
+                if (param.Interprete?.Exceptions == null)
+                {
+                    continue;
+                }
+
+                foreach (var exception in param.Interprete.Exceptions)
+                {
+                    if (exception.Display == null)
+                    {
+                        continue;
+                    }
+
+                    var state = exception.Display.State;
+                    (GenericStatus status, string rawDisplayValue, var stateValue) = GenericTests.CheckBasics(state, isRequired: true);
+
+                    // Missing
+                    if (status.HasFlag(GenericStatus.Missing))
+                    {
+                        results.Add(Error.MissingAttribute(this, param, exception.Display, param.Id?.RawValue).WithExtraData(ExtraData.ExceptionToFix, exception));
+                        continue;
+                    }
+
+                    // Empty
+                    if (status.HasFlag(GenericStatus.Empty))
+                    {
+                        results.Add(Error.EmptyAttribute(this, param, exception.Display, param.Id?.RawValue).WithExtraData(ExtraData.ExceptionToFix, exception));
+                        continue;
+                    }
+
+                    // Invalid
+                    if (status.HasFlag(GenericStatus.Invalid))
+                    {
+                        results.Add(Error.InvalidAttributeValue(this, param, exception.Display, exception.Display.State.RawValue, param.Id?.RawValue).WithExtraData(ExtraData.ExceptionToFix, exception));
+                        continue;
+                    }
+
+                    if (exception.Display.State.Value == Models.Protocol.Enums.EnumDisplayState.Enabled)
```

Comment:
exception.Display.State.Value can be replaced with the return value from the GenericTests.CheckBasics.
Also, put the namespace for the enums in the using list. This way the if isn't so long